ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -81,17 +87,17 @@ func init() {
 }
 
 type AWSClients struct {
-	ECSClient ecsclient.ECSClient
-	CFNClient cloudformation.CloudformationClient
-	SSMClient ssm.Client
+	ECSClient         ecsclient.ECSClient
+	CFNClient         cloudformation.CloudformationClient
+	AMIMetadataClient amimetadata.Client
 }
 
 func newAWSClients(commandConfig *config.CommandConfig) *AWSClients {
 	ecsClient := ecsclient.NewECSClient(commandConfig)
 	cfnClient := cloudformation.NewCloudformationClient(commandConfig)
-	ssmClient := ssm.NewSSMClient(commandConfig)
```

Comment:
If you don't feel like maintaining ssm code, you could look at this nifty cfn feature:

https://aws.amazon.com/blogs/mt/integrating-aws-cloudformation-with-aws-systems-manager-parameter-store/

Reply:
Lol, we maintain the SSM client just so that we can have this fancy info message:
```
INFO[0001] Using recommended Amazon Linux 2 AMI with ECS Agent 1.22.0 and Docker version 18.06.1-ce 
```